src/features/unified-share-modal/utils/mergeContacts.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+
+import type { contactType as Contact } from '../flowTypes';
+
+const mergeContacts = (existingContacts: Array<Contact>, fetchedContacts: Object): Array<Contact> => {
+    const contactsMap = Object.keys(fetchedContacts).reduce((map, email) => {
+        const contact = fetchedContacts[email];
+        map[email] = { ...contact, text: contact.name, value: contact.email || contact.id };
+        return map;
+    }, {});
+
+    return existingContacts.map(contact => {
+        if (contact.id) {
+            return contact;
+        }
+        return (
+            (contact.value && contactsMap[contact.value]) || {
+                email: String(contact.value),
+                id: 'not-previously-collaborated',
```

Comment:
this seems weird? wouldn't that cause a bunch of duplicate ids if a bunch of contacts were not found? this should probably be a unique identifier like the email. It's either that or return the original SelectOptionType

Reply:
Ok, I'll just use email